src/usePositionStickyObserver/usePositionStickyObserver.ts

```diff
@@ -0,0 +1,58 @@
+import { Ref, computed, onMounted, ref } from 'vue'
+import { MaybeRefOrGetter } from '@/types/maybe'
+import { useIntersectionObserver } from '@/useIntersectionObserver'
+import { toValue } from '@/utilities/vue'
+
+export type UsePositionStickyObserverResponse = {
+  stuck: Ref<boolean>,
+}
+
+export type UsePositionStickyObserverOptions = {
+  rootMargin?: string,
+  boundingElement?: MaybeRefOrGetter<HTMLElement | undefined>,
```

Comment:
I'd remove the MaybeRrfOrGetter type from this. If anyone needs to use a ref or getter they can pass a getter for the options themselves. 

Might simplify the options computed below. 

But doesn't hurt to keep it if you think it's needed.